app/assets/javascripts/web-dev-listener.js

```diff
@@ -1,5 +1,11 @@
 // TODO: don't serve this script from codecombat.com; serve it from a harmless extra domain we don't have yet.
 
+var lastSource = null;
+var lastOrigin = null;
+window.onerror = function(message, url, line, column, error){
+  console.log("Got an error!", arguments);
+  lastSource.postMessage({ type: 'error', message:message, url:url, line:line, column:column, error:error.toString() }, lastOrigin);
```

Comment:
Inconsistent `key: value` and `key:value` spacing. I like `key: value`, but whatever.